internal/clients/grpc/connect.go

```diff
@@ -45,20 +50,52 @@ func NewClientHandler(c *Client) ClientHandler {
 
 // Connect creates new gRPC client and connect to gRPC server.
 func Connect(cfg Config) (*gogrpc.ClientConn, bool, error) {
-	var opts []gogrpc.DialOption
+	opts := []gogrpc.DialOption{
+		gogrpc.WithUnaryInterceptor(otelgrpc.UnaryClientInterceptor()),
+	}
 	secure := false
 	tc := insecure.NewCredentials()
 
-	if cfg.ClientTLS && cfg.CACerts != "" {
-		var err error
-		tc, err = credentials.NewClientTLSFromFile(cfg.CACerts, "")
+	if cfg.ServerCAFile != "" {
+		tlsConfig := &tls.Config{}
+
+		// Loading System certificates
+		sysCertPool, err := x509.SystemCertPool()
 		if err != nil {
-			return nil, secure, err
+			fmt.Printf("failed to load system certificate %s\n", err.Error())
 		}
+		if sysCertPool != nil {
+			tlsConfig.RootCAs = sysCertPool
+		}
+
+		// Loading root ca certificates file
+		rootCA, err := loadCertFile(cfg.ServerCAFile)
```

Comment:
Since we check `if cfg.ServerCAFile != ""` on L59 the `cfg.ServerCAFile` can't be empty on L72 and hence L148. I'd suggest to remove `loadCertFile` and have something like 
```go
		// Loading root ca certificates file
		rootCA, err := ioutil.ReadFile(cfg.ServerCAFile)
		if err != nil {
			return nil, secure, fmt.Errorf("failed to read root ca file: %w", err)
		}
```

Reply:
Done